python/pyspark/sql/session.py

```diff
@@ -445,7 +487,12 @@ def _inferSchemaFromList(self, data, names=None):
             raise ValueError("Some of types cannot be determined after inferring")
         return schema
 
-    def _inferSchema(self, rdd, samplingRatio=None, names=None):
+    def _inferSchema(
+        self,
+        rdd: "RDD[Union[DateTimeLiteral, LiteralType, DecimalLiteral, RowLike]]",
```

Comment:
Just wondering about this ‒ I have a feeling that it should be either `RDD[Any]` (type-wise we can invoke this on arbitrary RDD) or, if we want to give a signal that can succeed  only on certain types of RDDs,  `Literal*` variants should be omitted (we don't support schema inference on these).

Same applies to `_inferSchemaFromList`.

Reply:
Should we use `Any` from `createDataFrame` then?